src/main/kotlin/com/jfrog/conan/clionplugin/conan/ConanPluginUtils.kt

```diff
@@ -0,0 +1,46 @@
+package com.jfrog.conan.clionplugin.conan
+
+import com.intellij.openapi.diagnostic.thisLogger
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+import java.io.File
+import java.net.URL
+import java.nio.file.Paths
+
+object ConanPluginUtils {
+    private const val cmakeProviderFileName = "conan_provider.cmake"
+    private const val cmakeProviderURL = "https://raw.githubusercontent.com/conan-io/cmake-conan/develop2/conan_provider.cmake"
+
+    fun getPluginHome(): String {
+        return Paths.get(System.getProperty("user.home"), ".conan-clion-plugin").toString()
+    }
+
+    fun downloadCMakeProvider(update: Boolean = false) {
+        val pluginHome = getPluginHome()
+        val targetFile = File(pluginHome, cmakeProviderFileName)
+
+        if (update || !targetFile.exists()) {
+            targetFile.parentFile.mkdirs()
+            targetFile.downloadFromUrl(cmakeProviderURL)
+        }
+    }
+
+    private fun File.downloadFromUrl(url: String) {
```

Comment:
I'd move this to `conan/extensions/file.kt` to keep the extensions available to other parts of the code if needed